apps/prairielearn/src/server.js

```diff
@@ -165,6 +165,20 @@ module.exports.initExpress = function () {
     }),
   );
 
+  app.use((req, res, next) => {
+    const issuedAt = new Date(req.session.cookie.expires - req.session.cookie.originalMaxAge);
+
+    // If the cookie was issued more than an hour in the past, refresh it
+    // with a new expiration date.
+    if (issuedAt < new Date(Date.now() - 60 * 60 * 1000)) {
```

Comment:
This little bit of math means that if someone authenticates, makes requests for 59 minutes, and then does nothing, we'll expire their session sooner than `config.sessionStoreExpireSeconds` after their last request. I think this is fine, but we could also decrease this threshold if we wanted. At any rate, I imagine we'll want to increase this if we start using sessions in place of the `pl_authn` cookie.

Reply:
This could be a config, and you could have the actual expiry be added to this value (e.g., if the config says expire after 6 hours, actually you expire after 7). In that case the expiry could be longer than configured, but I think that could be more acceptable than the alternative.

Reply:
On second thought: that might not work how we want. Even if we have the backend keep sessions alive for that extra amount of time, the cookie will still expire before the backend session will. We could add this same delta to the cookie expiration time? But at that point, we might just pick a value for `sessionStoreExpireSeconds` that's large enough compared to this delta that the cookie/session expiring slightly early won't make a perceptable difference to the user. That is, is sessions last 14 days, they probably won't care if it actually expires 13 days and 23 hours since their last activity (since they've already been inactive that long to begin with).